templates/openstackconfiggenerator/bin/create-playbooks.sh

```diff
@@ -85,6 +85,28 @@ openstack tripleo container image prepare $PREPARE_ENV_ARGS -r roles_data.yaml -
 mkdir -p $HOME/tripleo-deploy
 rm -rf $HOME/tripleo-deploy/overcloud-ansible*
 
+{{- if eq .OSPVersion "16.2" }}
+# Replicate the (undocumented) heat client merging of map parameters
+# STF, Contrail, Trilio and presumably others currently rely on it
+/home/cloud-admin/process-heat-environment.py \
+    $TEMPLATES_DIR/overcloud-resource-registry-puppet.yaml \
+    $TEMPLATES_DIR/tripleo-overcloud-images.yaml \
+    $TEMPLATES_DIR/environments/deployed-server-environment.yaml \
+    $TEMPLATES_DIR/environments/docker-ha.yaml \
+    $TEMPLATES_DIR/environments/network-isolation.yaml \
+    $TEMPLATES_DIR/environments/network-environment.yaml \
+{{- range $i, $value := .TripleoEnvironmentFiles }}
+    $TEMPLATES_DIR/{{ $value }} \
+{{- end }}
+{{- range $key, $value := .TripleoDeployFiles }}
+    {{ $key }} \
+{{- end }}
+{{- range $key, $value := .TripleoCustomDeployFiles }}
+    {{ $key }} \
+{{- end }}
```

Comment:
looks good to me, just one clarification, should this environment list be always that same as in the stack create (except the password file)? Just thinking if it would be good if the script also uses -e for any parameter file and then  just create one var containing all env files and use it for both process-heat-env and stack create, like  PREPARE_ENV_ARGS. So that its only required to do any change in a single place.

Reply:
yeah, could do that